cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/OperationalLimitConversion.java

```diff
@@ -73,107 +80,143 @@ private void setVoltageLevelForVoltageLimit(Terminal terminal) {
-        }
-        switch (limitSubClass) {
-            case ACTIVE_POWER_LIMIT:
-                return b::newActivePowerLimits1;
-            case APPARENT_POWER_LIMIT:
-                return b::newApparentPowerLimits1;
-            case CURRENT_LIMIT:
-                return b::newCurrentLimits1;
-            default:
-                throw new IllegalStateException();
-        }
-    }
-
-    private static Supplier<LoadingLimitsAdder<?, ?>> getLoadingLimitAdder2Supplier(String limitSubClass, Branch<?> b) {
+    /**
+     * Get the LoadingLimitsAdder supplier for the given limits group and subclass.
+     * @param limitsGroup The limit group instance for which the adder is called.
+     * @param limitSubClass The subclass of the OperationalLimit.
+     * @return The appropriate LoadingLimitsAdder supplier.
+     */
+    private Supplier<LoadingLimitsAdder<?, ?>> getLoadingLimitAdderSupplier(OperationalLimitsGroup limitsGroup, String limitSubClass) {
         if (limitSubClass == null) {
-            return b::newCurrentLimits2;
+            return limitsGroup::newCurrentLimits;
         }
         switch (limitSubClass) {
             case ACTIVE_POWER_LIMIT:
-                return b::newActivePowerLimits2;
+                return limitsGroup::newActivePowerLimits;
             case APPARENT_POWER_LIMIT:
-                return b::newApparentPowerLimits2;
+                return limitsGroup::newApparentPowerLimits;
             case CURRENT_LIMIT:
-                return b::newCurrentLimits2;
+                return limitsGroup::newCurrentLimits;
             default:
-                throw new IllegalStateException();
+                throw new IllegalArgumentException();
         }
     }
 
-    private void createLimitsAdder(int terminalNumber, String limitSubClass, Branch<?> b) {
+    /**
+     * Create the LoadingLimitsAdder for the given branch + side and the given limit set + subclass.
+     * @param terminalNumber The side of the branch to which the OperationalLimit applies.
+     * @param limitSubClass The subclass of the OperationalLimit.
+     * @param limitSetId The id of the set containing the OperationalLimit.
+     * @param limitSetName The name of the set containing the OperationalLimit.
+     * @param b The branch to which the OperationalLimit applies.
+     */
+    private void createLimitsAdder(int terminalNumber, String limitSubClass, String limitSetId, String limitSetName, Branch<?> b) {
         if (terminalNumber == 1) {
-            Supplier<LoadingLimitsAdder<?, ?>> loadingLimitAdder1Supplier = getLoadingLimitAdder1Supplier(limitSubClass, b);
-            loadingLimitsAdder1 = context.loadingLimitsMapping().computeIfAbsentLoadingLimitsAdder(b.getId() + "_1_" + limitSubClass, loadingLimitAdder1Supplier);
+            OperationalLimitsGroup limitsGroup = b.getOperationalLimitsGroup1(limitSetId).orElseGet(() -> {
+                b.setProperty(PROPERTY_PREFIX + limitSetId, limitSetName);
```

Comment:
I agree it would make much more sense to attach the id and the name in two different properties with a unique key to the OperationalLimitsGroup, but this is sadly not possible since OperationalLimitsGroup doesn't inherit Identifiable (which has the methods to set/get properties). That's why these properties have been attached to the Branch/DanglingLine/ThreeWindingTransformer parents.
Also, since one of these parents can have multiple OperationalLimitsGroup, it's not possible to use a unique key and that's how I came up to create a property with this key/value pair.

Reply:
yes, indeed :cry: Well good thing it's temporary until a new iidm version